src/legacy/core_plugins/kibana/public/dashboard/panel/dashboard_panel.tsx

```diff
@@ -17,127 +17,162 @@
  * under the License.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import { injectI18n, FormattedMessage } from '@kbn/i18n/react';
+import { FormattedMessage, injectI18n } from '@kbn/i18n/react';
+
+import { EuiPanel } from '@elastic/eui';
 import classNames from 'classnames';
 import _ from 'lodash';
-
-import { PanelHeader } from './panel_header';
+import React from 'react';
+import {
+  ContainerState,
+  Embeddable,
+  EmbeddableFactory,
+  EmbeddableMetadata,
+  EmbeddableState,
+} from 'ui/embeddable';
+import { EmbeddableActions, EmbeddableErrorAction } from '../actions';
+import { PanelState } from '../selectors';
 import { PanelError } from './panel_error';
+import { PanelHeader } from './panel_header/panel_header';
+
+interface DashboardPanelProps {
+  intl: any;
+  viewOnlyMode: boolean;
+  onPanelFocused: (panel: string) => {};
+  onPanelBlurred: (panel: string) => {};
+  error: string | object;
+  destroy: () => void;
+  containerState: ContainerState;
+  embeddableFactory: EmbeddableFactory;
+  embeddableStateChanged: (embeddableStateChanges: EmbeddableState) => {};
+  embeddableIsInitializing: EmbeddableActions;
+  embeddableIsInitialized: (embeddableIsInitialized: EmbeddableMetadata) => void;
+  embeddableError: EmbeddableErrorAction;
+  initialized: boolean;
+  panel: PanelState;
+  isPopoverOpen: boolean;
+  isLoading: boolean;
+  calloutTitle: string;
+  className: any;
+}
 
-import {
-  EuiPanel,
-} from '@elastic/eui';
+interface State {
+  error: string | null;
+}
 
-class DashboardPanelUi extends React.Component {
-  constructor(props) {
+export class DashboardPanelUi extends React.Component<DashboardPanelProps, State> {
+  [panel: string]: any;
+  public mounted: boolean;
+  public embeddable!: Embeddable;
+  constructor(props: DashboardPanelProps) {
     super(props);
     this.state = {
-      error: props.embeddableFactory ? null : props.intl.formatMessage({
-        id: 'kbn.dashboard.panel.noEmbeddableFactoryErrorMessage',
-        defaultMessage: 'No factory found for embeddable',
-      }),
+      error: props.embeddableFactory
+        ? null
+        : props.intl.formatMessage({
+            id: 'kbn.dashboard.panel.noEmbeddableFactoryErrorMessage',
+            defaultMessage: 'No factory found for embeddable',
+          }),
     };
-
     this.mounted = false;
   }
 
-  async componentDidMount() {
+  public async componentDidMount() {
     this.mounted = true;
     const {
       initialized,
       embeddableFactory,
-      embeddableIsInitializing,
       panel,
       embeddableStateChanged,
       embeddableIsInitialized,
-      embeddableError,
     } = this.props;
-
     if (!initialized) {
-      embeddableIsInitializing();
-      embeddableFactory.create(panel, embeddableStateChanged)
-        .then((embeddable) => {
+      embeddableFactory
```

Comment:
Looks like you dropped the first call to `embeddableIsInitializing();`.  intentional?  Might be the reason for the failing tests.

Reply:
Thanks a ton! I was struggling on an error when adding this back in but recently figured that out. Changed the code to include embeddableIsInitializing() and changed line 48 to   `embeddableIsInitializing: () => void;`